config/git_fetcher.go

```diff
@@ -126,6 +126,31 @@ func (g *GitFetcher) Get(key string) (val string, ok bool) {
 	return
 }
 
+func (g *GitFetcher) All() map[string]string {
+	newmap := make(map[string]string)
+
+	g.vmu.RLock()
+	defer g.vmu.RUnlock()
+
+	for key, value := range g.vals {
+		newmap[key] = value
+	}
+
+	return newmap
+}
+
+func (g *GitFetcher) Set(key, value string) {
+	g.vmu.RLock()
+	defer g.vmu.RUnlock()
+	g.vals[strings.ToLower(key)] = value
+}
+
+func (g *GitFetcher) Del(key string) {
+	g.vmu.RLock()
```

Comment:
Since this is a write operation, shouldn't we use the `Lock()` and `Unlock()` methods, since `RLock` and `RUnlock` are designated to readers?